### PR TITLE
Improve workflow selection for anonymous users

### PR DESCRIPTION
**Pull request by eatyourgreens**
Staging branch URL: https://pr-5107.pfe-preview.zooniverse.org

- Changes the project home page workflow links so that they defer loading the classify page until the new workflow has loaded.
- Changes workflow selection so that a new workflow is not loaded if there is already an active workflow for the project on mount.
- Updates components that load workflows so that translations are loaded in parallel with workflows, rather than after workflows load.

# Required Manual Testing

- [x] Does the non-logged in home page render correctly?
- [x] Does the logged in home page render correctly?
- [x] Does the projects page render correctly?
- [x] Can you load project home pages?
- [ ] Can you load the classification page?
- [ ] Can you submit a classification?
- [ ] Does talk load correctly?
- [ ] Can you post a talk comment?

# Review Checklist

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `rm -rf node_modules/ && npm install` and app works as expected?
- [ ] If the component is in coffeescript, is it converted to ES6? Is it free of eslint errors? Is the conversion its own commit?
- [x] Are the tests passing locally and on Travis?

# Optional

- [ ] Have you replaced any `ChangeListener` or `PromiseRenderer` components with code that updates component state?
- [ ] If changes are made to the classifier, does the dev classifier still work?
- [ ] Have you [resized and compressed](https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/image-optimization) any image you've added?
- [ ] Have you added in [flow type annotations](https://flowtype.org/docs/type-annotations.html)?
- [ ] Have you followed the [Springer guidelines for commit messages](https://github.com/springernature/frontend-playbook/blob/master/git/git.md#commit-messages)?

### CODE REVIEW

File: app/pages/project/workflow-selection.jsx [hunk 1]
```diff
@@ -12,12 +12,17 @@ class WorkflowSelection extends React.Component {
     super();
     this.state = {
       error: null,
-      loadingSelectedWorkflow: true
+      loadingSelectedWorkflow: false
```
Comment (wgranger) [inline app/pages/project/workflow-selection.jsx]: Why was this changed to false? It seems with this change the component would load on initial render https://github.com/zooniverse/Panoptes-Front-End/blob/97f8bc24ba926fa24ab6756b82727cd0d8efbd16/app/pages/project/workflow-selection.jsx#L190

Comment (eatyourgreens) [reply app/pages/project/workflow-selection.jsx]: If you set this to true, you should notice that the classifier never loads.

Comment (wgranger) [reply app/pages/project/workflow-selection.jsx]: Ahh. You are correct.

Comment (eatyourgreens) [reply app/pages/project/workflow-selection.jsx]: If there’s already a valid workflow loaded, we set this to false and skip workflow loading completely.

Comment (eatyourgreens) [reply app/pages/project/workflow-selection.jsx]: That could be set explicitly in `componentDidMount` if that makes the code easier to understand. I’ll leave it to you to change if you think that’s better.

File: app/pages/project/workflow-selection.jsx [hunk 2]
```diff
@@ -12,12 +12,17 @@ class WorkflowSelection extends React.Component {
...
     };
   }
 
   componentDidMount() {
-    this.getSelectedWorkflow(this.props);
+    const { project, workflow } = this.props;
+    const linkedActiveWorkflows = project.links.active_workflows || [];
+    const workflowExistsForProject = workflow && linkedActiveWorkflows.indexOf(workflow.id) > -1;
+    if (!workflowExistsForProject) {
+      this.getSelectedWorkflow(this.props);
```
Comment (wgranger) [inline app/pages/project/workflow-selection.jsx]: Ok, so this only gets called if it needs to (if there isn't currently a workflow present? And this is what eliminates the second workflow request?

File: app/pages/project/workflow-selection.jsx (hunk 1 above)
Comment (wgranger) [reply app/pages/project/workflow-selection.jsx]: No, I think it's unnecessary to have an additional setState in `componentDidMount` if this is the property you want.

**Comment by wgranger** [comment]
I've tested this PR on a majority of the active projects, signed in and not, and they all appear to handle workflow selection correctly.

File: app/pages/project/workflow-selection.jsx (hunk 2 above)
Comment (eatyourgreens) [reply app/pages/project/workflow-selection.jsx]: Yes. Without this change, a workflow is always loaded on component mount, even if there’s already one in the Redux store.
